js/walutomat.js

```diff
@@ -0,0 +1,274 @@
+                'INVALID PARAMETER': ExchangeError,
+                'INSUFFICIENT_FUNDS': InsufficientFunds,
+                'MISSING_API_KEY': AuthenticationError,
+            },
+        });
+    }
+
+    fetchMarkets (params = {}) {
+        const currencies = ['EUR', 'GBP', 'USD', 'CHF', 'PLN'];
+        const markets = [];
+        const currenciesLength = currencies.length;
+        for (let base = 0; base < currenciesLength - 1; base++) {
+            const newIndex = base + 1;
+            for (let quote = newIndex; quote < currencies.length; quote++) {
+                markets.push ({
+                    'id': currencies[base] + '_' + currencies[quote],
+                    'symbol': currencies[base] + '/' + currencies[quote],
+                    'base': currencies[base],
+                    'quote': currencies[quote],
+                    'active': true,
+                    'precision': {
+                        'amount': 2,
+                        'price': 4,
+                    },
+                    'info': currencies[base] + ' - ' + currencies[quote] + ' exchange',
+                });
+            }
+        }
+        return markets;
+    }
+
+    async fetchBalance (params = {}) {
+        const balances = await this.privateGetAccountBalances ();
+        const result = {};
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            result[this.safeString (balance, 'currency')] = {
+                'free': this.safeFloat (balance, 'balanceAvailable'),
+                'used': this.safeFloat (balance, 'balanceReserved'),
+                'total': this.safeFloat (balance, 'balanceAll'),
+            };
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const symbolObject = { 'symbol': this.marketId (symbol) };
+        const response = await this.publicGetPublicMarketOrderbookSymbol (this.extend (symbolObject, params));
+        return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'baseVolume');
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            'operationType': 'MARKET_FX',
+            'sortOrder': 'DESC',
+        };
+        if (symbol !== undefined) {
+            request['currencies'] = this.marketId (symbol);
+        }
+        if (since !== undefined) {
+            // TODO: Not sure if it's correct (idk if it's timestamp or DateTime string
+            request['continueFrom'] = since;
+        }
+        if (limit !== undefined) {
+            request['volume'] = limit;
+        }
+        const response = await this.privateGetAccountHistory (this.extend (request, params));
+        const result = [];
+        for (let i = 0; i < response.length; i++) {
```

Comment:
Ok, a few more edits, namely, this particular loop is already generalized to use `parseTrades`.

See `fetchMyTrades` implementation here for example (or, practially any implementation of that method):

- https://github.com/ccxt/ccxt/blob/master/js/hitbtc2.js#L1231

↑ It calls `parseTrades` that loops over the results and applies `parseTrade` to each one of them and then returns the parsed trades. Can you plz add this change here? We're almost there and will merge this as soon as we have most of these minor issues cleaned out, there's not so many of them left, i think. Thx again for your involvement and for your understanding – really appreciate that!